.github/workflows/code.yml

```
@@ -91,13 +99,13 @@ jobs:
         timeout-minutes: 10
 
       - name: Build binaries
-        run: cargo build --release --locked
+        run: cargo build --release --locked ${{ matrix.test.build-args }}
 
       - name: Run bash e2e tests
-        run: utils/e2e-tests/bash/run-tests "target/release/humanode-peer"
+        run: utils/e2e-tests/bash/run-tests ${{ matrix.test.suite }} "target/release/humanode-peer"
```

```suggestion
        run: utils/e2e-tests/bash/run-tests "$TEST_SUITE" "target/release/humanode-peer"
        env:
           TEST_SUITE: ${{ matrix.test.suite }}
```

This is a bit safer if we consider quoting and shell injection

---

Got it, thanks, improved

---

You got it slightly wong though - see the quoting around the env var